chacractl/api/binaries.py

```diff
@@ -71,27 +87,36 @@ def post(self, url, filepath):
             return self.put(file_url, filepath)
         elif exists.status_code == 404:
             logger.info('POSTing file: %s', filepath)
-            with open(filepath, 'rb') as binary:
-                response = requests.post(
+            binary, digest = self.load_file(filepath)
+            response = requests.post(
                     url,
                     files={'file': binary},
                     auth=chacractl.config['credentials'],
                     verify=chacractl.config['ssl_verify'])
         if response.status_code > 201:
             logger.warning("%s -> %s", response.status_code, response.text)
             response.raise_for_status()
+        if not self.verify_upload(url, filename, digest):
+            # Since this is a new file, attempt to delete it
```

Comment:
Let's log what happened here before calling `delete()`. If that `delete()` call were to fail there, it would not be immediately clear why chacractl was attempting a DELETE at all.

We can also print `filename` in the log "Uploaded checksum mismatch" messages.